src/target/riscv/riscv.c

```diff
@@ -4196,6 +4196,19 @@ COMMAND_HANDLER(handle_info)
 		riscv_enumerate_triggers(target) == ERROR_OK;
 	riscv_print_info_line_if_available(CMD, "hart", "trigger_count",
 				r->trigger_count, trigger_count_available);
+	/* dcsr.ebreak flags status */
+	riscv_print_info_line(CMD, "OpenOCD_flag_dcsr", "ebreakm", riscv_ebreakm);
+	riscv_print_info_line(CMD, "OpenOCD_flag_dcsr", "ebreaks", riscv_ebreaks);
+	riscv_print_info_line(CMD, "OpenOCD_flag_dcsr", "ebreaku", riscv_ebreaku);
+
+	riscv_reg_t dcsr;
+
+	const bool dcsr_read_success = riscv_get_register(target, &dcsr, GDB_REGNO_DCSR) == ERROR_OK;
+
+	riscv_print_info_line_if_available(CMD, "dcsr", "ebreakm", get_field(dcsr, CSR_DCSR_EBREAKM), dcsr_read_success);
+	riscv_print_info_line_if_available(CMD, "dcsr", "ebreaks", get_field(dcsr, CSR_DCSR_EBREAKS), dcsr_read_success);
+	riscv_print_info_line_if_available(CMD, "dcsr", "ebreaku", get_field(dcsr, CSR_DCSR_EBREAKU), dcsr_read_success);
```

Comment:
Is the purpose of these fields to tell you the current state of these `dcsr.ebreakX` bits?

Why do we need to add them to the output? Why not to read the value of `dcsr` directly using one of the usual register read commands?

Reply:
Firstly, I want to add current state of OpenOCD `riscv_ebreak*` flags. After that, I thought that user can be a little confused by these values (because there is an active dcsr state and the one that will be set upon next resume), so I decided to add actual `dcsr.ebreak*` values to `riscv info` output.

> Why not to read the value of dcsr directly using one of the usual register read commands?

Now it has become much easier to check this values.